src/react/when.tsx

```diff
@@ -0,0 +1,40 @@
+import Policy from '../policy';
+import { usePundit } from './pundit-provider';
+
+interface WhenProps {
+  children: JSX.Element | null;
+  can: string;
+  policy?: Policy;
+  user?: unknown;
+  record?: unknown;
+}
+
+export default function When({
+  children,
+  can,
+  policy,
+  user,
+  record,
+}: WhenProps): JSX.Element | null {
+  const { policy: hookPolicy } = usePundit();
+  let canPerformAction = false;
+  if (policy) {
```

Comment:
I'm wondering if there's a slightly better way to go about this. Particularly the mutation here makes me weary. Maybe the policy itself can have some sort of duplicate method where it merges in a new user/record?

It could clean up the logic here a bit, too 🤔 

```ts
const paramPolicy = policy.copy(user, record)
```

Or, perhaps the `usePundit` hook can automagically do that for us, too. Anyway, I'm just spitballing here. What are your thoughts?

Reply:
That's a good idea adding a copy method. I've pushed up an amendment adding this and removing the mutation, so the logic is less sprawling within the `When` function now.

Reply:
Dig it! 😎 🎉